components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt

```diff
@@ -267,7 +269,13 @@ class GeckoEngine(
             onSuccess(updatedExtension)
             GeckoResult<Void>()
         }, { throwable ->
-            onError(extension.id, throwable)
+
+            val exception = if (throwable is InstallException && throwable.code != ERROR_NETWORK_FAILURE) {
```

Comment:
Looks like ERROR_USER_CANCELED and ERROR_POSTPONED should also be treated as recoverable? For the former we don't have UI, but seems worth handling regardless.